### PR TITLE
CLDR-15637 add checks on consistency of modifiers

### PR DESCRIPTION
**Pull request by macchiati**
Add checks on consistency of modifiers
Add checks on consistency of values for -core, -prefix, and plain

CLDR-15637

- [ ] This PR completes the ticket.

### CODE REVIEW

**Review by pedberg-icu — DISMISSED** [review]

File: tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckPlaceHolders.java [hunk 1]
```diff
@@ -135,13 +150,31 @@ public CheckCLDR handleCheck(String path, String fullPath, String value, Options
...
                     }
+                } else { // real value
+                    // special checks for prefix/core
+                    final boolean hasPrefix = modifiers.contains(Modifier.prefix);
+                    final boolean hasCore = modifiers.contains(Modifier.core);
+                    if (hasPrefix || hasCore) {
+                        // We need consistency among the 3 values if we have either prefix or core
+
+                        String coreValue = hasCore ? value : modifiedFieldValue(parts, field, modifiers, Modifier.core);
+                        String prefixValue = hasCore ? value : modifiedFieldValue(parts, field, modifiers, Modifier.prefix);
```
Comment (richgillam) [inline tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckPlaceHolders.java:166]: Should `hasCore` be `hasPrefix` here?

**Comment by macchiati** [comment]
good catch! I'll make a PR for that.

**Review by richgillam — DISMISSED** [review]

**Comment by macchiati** [comment]
there was no material change from the last instance; just had to rebase and run the tests.

**Review by pedberg-icu — APPROVED** [review]

File: tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckPlaceHolders.java (hunk 1 above)
Comment (pedberg-icu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckPlaceHolders.java]: I see that was later fixed.

**Pull request merged**